cmd/gctsCreateRepository.go

```diff
@@ -0,0 +1,156 @@
+		GithubURLstring string `json:"url"`
+	}
+
+	type createRequestBody struct {
+		Repository string   `json:"repository"`
+		Data       repoData `json:"data"`
+	}
+
+	type repoConfig struct {
+		Key      string `json:"key"`
+		Value    string `json:"value"`
+		Category string `json:"category"`
+	}
+
+	type createResultBody struct {
+		RID         string       `json:"rid"`
+		Name        string       `json:"name"`
+		Role        string       `json:"role"`
+		Type        string       `json:"type"`
+		VSID        string       `json:"vsid"`
+		Status      string       `json:"status"`
+		Branch      string       `json:"branch"`
+		URL         string       `json:"url"`
+		CreatedBy   string       `json:"createdBy"`
+		CreatedDate string       `json:"createdDate"`
+		Connection  string       `json:"connection"`
+		Config      []repoConfig `json:"config"`
+	}
+
+	type createResponseBody struct {
+		Repository createResultBody `json:"repository"`
+		Exception  string           `json:"exception"`
+	}
+
+	reqBody := createRequestBody{
+		Repository: config.Repository,
+		Data: repoData{
+			RID:             config.Repository,
+			Name:            config.Repository,
+			Role:            config.Role,
+			Type:            config.Type,
+			VSID:            config.VSID,
+			GithubURLstring: config.RemoteRepositoryURL,
+		},
+	}
+	jsonBody, marshalErr := json.Marshal(reqBody)
+
+	if marshalErr != nil {
+		return fmt.Errorf("creating the repository locally failed: %w", marshalErr)
+	}
+
+	header := make(http.Header)
+	header.Set("Content-Type", "application/json")
+	header.Add("Accept", "application/json")
+
+	url := "http://" + config.Host +
+		"/sap/bc/cts_abapvcs/repository?sap-client=" + config.Client
+
+	resp, httpErr := httpClient.SendRequest("POST", url, bytes.NewBuffer(jsonBody), header, nil)
+
+	defer func() {
+		if resp != nil && resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	if resp == nil {
+		return fmt.Errorf("creating the repository locally failed: %w", httpErr)
+	}
+
+	var response createResponseBody
+	parsingErr := parseHTTPResponseBodyJSON(resp, &response)
+
+	if parsingErr != nil {
+		log.Entry().Warning(parsingErr)
+	}
+
+	if httpErr != nil {
+		if resp.StatusCode == 500 && response.Exception == "Repository already exists" {
+			log.Entry().
+				WithField("repository", config.Repository).
+				Info("the repository already exists locally")
+			return nil
+		}
+		return fmt.Errorf("creating the repository locally failed: %w", httpErr)
+	}
+
+	log.Entry().
+		WithField("repository", config.Repository).
+		Info("successfully created the local repository")
+	return nil
+}
+
+func parseHTTPResponseBodyJSON(resp *http.Response, response interface{}) error {
+	if resp == nil {
+		return fmt.Errorf("cannot parse HTTP response with value <nil>")
+	}
+	bodyText, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("could not read HTTP response body: %w", err)
+	}
+	json.Unmarshal(bodyText, &response)
```

Comment:
Any error raised by `json.Unmarshal` is swapped under the carpet.

In case we have a 4xx or 5xx we have most likely nothing we can parse to json, but the body content of the response might be highly valuable in order to troubleshoot the issue. I would like to suggest to have in that case the body content plain in the log. Only case where this does not apply is when we have to expect some secrets (access token) in the response body. If that is expected to be the case might depend on the http response code.

In case there is a **real** good reason for swallowing the error here I would like to make that explict via `_ = json.Unmarshal(./.)` alongside with a comment explaining the why.

Reply:
There is no good reason to neglect the error. Will fix.